code/modules/modular_computers/hardware/_hardware.dm

```diff
@@ -87,3 +87,32 @@
 	damage += round(amount) 					// We want nice rounded numbers here.
 	damage = between(0, damage, max_damage)		// Clamp the value.
 
+// This is called whenever hardware is inserted into a modular computer.
+// Currently only has a use for 'manual' flash drives
+/obj/item/weapon/computer_hardware/proc/installed(var/obj/item/modular_computer/M)
+	if(istype(src, /obj/item/weapon/computer_hardware/hard_drive/portable/manual))
```

Comment:
this is what inheritance is for. Have the special drive itself override this file and do its own thing